Replicator/Puller.cc

```diff
@@ -356,21 +356,59 @@ namespace litecore::repl {
         }
     }
 
-    Worker::ActivityLevel Puller::computeActivityLevel() const {
-        ActivityLevel level;
+    Worker::ActivityLevel Puller::computeActivityLevel(std::string* reason) const {
+        ActivityLevel         level;
+        int                   k = -1;
+        std::string           reason0;
+        Worker::ActivityLevel workerLevel{kC4Stopped};
         if ( _unfinishedIncomingRevs + _unfinishedIncomingRevoked > 0 ) {
             // CBL-221: Crash when scheduling document ended events
             level = kC4Busy;
+            k     = 0;
```

Comment:
Instead of this why not just set the reason here if applicable?

Reply:
I designed it in two stages. The first stage record the decision place. The goal is to not alter the current logical flow. The second stage, merely assigns reason to the final surviving k. (k can be overwritten later).

Reply:
That seems needlessly complex.  What is this requirement of "not altering the current code"?  The side effect is making the code harder to understand.